### ISSUE TITLE
Minor threading bug in HealthCheckScheduler

### ISSUE THREAD

**Issue by accwebs** [dropwizard/dropwizard-health#21]
I was doing a review of this library and noticed what I think is a minor threading bug in HealthCheckScheduler.

Specifically, the 'futures' member is declared to be a regular HashMap:
````
    private final Map<String, ScheduledFuture> futures = new HashMap<>();
````

But, the futures collection is modified by both the main thread:

````
HealthCheckRegistry.register() ->
   HealthCheckManager.onHealthCheckAdded() -> 
      HealthCheckScheduler.schedule() -> 
          modifies collection
````

as well as the health check threads created by the executor:

````
HealthCheckManger.onStateChanged() -> 
   HealthCheckScheduler.schedule() -> 
      HealthCheckScheduler.unschedule() ->
          modifies collection
````

Thus I think the futures member needs to be a synchronized collection (probably ConcurrentHashMap or Collections.synchronizedMap()) in order to ensure changes to the collection are published to other threads immediately.  Otherwise you might not be able to remove an entry already in the collection due to the lack of the health check's thread's ability to see the addition made by the main thread.

Likely this is very minor.

**Comment by mzamani1** [comment]
Thanks for reporting this @accwebs! I opened the following PR to fix this: https://github.com/dropwizard/dropwizard-health/pull/22

**Comment by mzamani1** [comment]
Released `1.3.16-1` containing a fix for this.

**Comment by accwebs** [comment]
Super service!  Thanks!
